src/processor-utils.js

```diff
@@ -355,11 +356,32 @@ function computeEffectiveGasPrice(gasPrice, effectivePercentage) {
  * @returns computed l2 tx hash
  */
 async function computeL2TxHash(tx) {
-    const txHash = await smtUtils.linearPoseidon(`0x${tx.nonce.slice(2)}${tx.gasPrice.slice(2)}${tx.gasLimit.slice(2)}${tx.to.slice(2)}${tx.value.slice(2)}${tx.data.slice(2)}${tx.from.slice(2)}`);
+    const hash = `${formatL2TxHashParam(tx.nonce)}${formatL2TxHashParam(tx.gasPrice)}${formatL2TxHashParam(tx.gasLimit)}${formatL2TxHashParam(tx.to)}${formatL2TxHashParam(tx.value)}${formatL2TxHashParam(tx.data)}${formatL2TxHashParam(tx.from)}`;
+    const txHash = await smtUtils.linearPoseidon(hash);
 
     return txHash;
 }
 
+function formatL2TxHashParam(param) {
```

Comment:
- missing unit test
- missing comments on function
- check with regex that is hex string